src/wp-includes/post.php

```diff
@@ -2289,7 +2296,21 @@ function add_post_type_support( $post_type, $feature, ...$args ) {
 	$features = (array) $feature;
 	foreach ( $features as $feature ) {
 		if ( $args ) {
-			$_wp_post_type_features[ $post_type ][ $feature ] = $args;
+			// Check if feature already exists with args and if both are associative arrays that should be merged.
+			if ( isset( $_wp_post_type_features[ $post_type ][ $feature ] )
+				&& is_array( $_wp_post_type_features[ $post_type ][ $feature ] )
+				&& isset( $_wp_post_type_features[ $post_type ][ $feature ][0] )
+				&& is_array( $_wp_post_type_features[ $post_type ][ $feature ][0] )
```

Comment:
```suggestion
			if ( isset( $_wp_post_type_features[ $post_type ][ $feature ][0] )
				&& is_array( $_wp_post_type_features[ $post_type ][ $feature ][0] )
```

I think we can simplify this check - https://3v4l.org/NIrMG.

Reply:
Good catch :) Applied in [0f59639](https://github.com/WordPress/wordpress-develop/pull/10426/commits/0f59639b43b94e5b03da2b1f0352099026db4f6e)